src/components/MessageInput.js

```diff
@@ -328,6 +338,16 @@ class MessageInput extends PureComponent {
   }
 
   componentDidUpdate(prevProps) {
+    if (Object.keys(this.state.asyncUploads).length) {
+      /**
+       * When successful image upload response occurs after hitting send,
+       * send a follow up message with the image
+       */
+      this.setState({ sending: true });
```

Comment:
Hey @DanC5 , can we move this state variable to class variable (`this.sending`). Since its not used on UI side, it can result in un-necessary re-rendering when this values updates. Also setState doesn't ensure updating of state immediately ... so that can also result in some issues!

Reply:
> Hey @DanC5 , can we move this state variable to class variable (`this.sending`). Since its not used on UI side, it can result in un-necessary re-rendering when this values updates. Also setState doesn't ensure updating of state immediately ... so that can also result in some issues!

@vishalnarkhede updated, let me know if you have other comments!